### ISSUE TITLE
fix(operators): linear operators are 2/3-masked but `inverse_laplacian`/`project_vector` are not, so `laplacian(inverse_laplacian(u)) != u` on the default grid

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#91]
Part of #80.

## Summary

With the default `dealias="2/3"`, `laplacian`, `gradient`, `biharmonic`, `hyperviscosity`, `divergence`, `curl` and `velocity_from_streamfunction` multiply by the 2/3 mask (via `k_dealias` in 1D and the mask in 2D/3D) before returning, so they zero every resolved mode with `N/3 < |k| ≤ N/2`. `inverse_laplacian` and `project_vector` do not. The result is that the two halves of the same operator pair disagree, and that a *linear* operator silently discards a third of the spectrum. Dealiasing is a property of products, not of derivatives.

This is why two notebooks bypass the public API: `docs/notebooks/navier_stokes_2d.py:143-160` and `qg_model.py:164-181` hand-roll the diffusion term with `grid.K2` and a comment about NaN, while `demo_qg.py:513-516` loops `deriv.laplacian`. The examples disagree with each other about how to use the package.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp
from spectraldiffx import FourierGrid1D, SpectralDerivative1D

g = FourierGrid1D.from_N_L(N=64, L=2 * jnp.pi)        # default dealias="2/3"
d = SpectralDerivative1D(grid=g)
u = jnp.cos(25 * g.x)                                  # resolved: 25 < N/2 = 32, but 25 > N/3
print(float(jnp.abs(d.laplacian(u)).max()))            # 1.9e-12   (exact: 625)
print(float(jnp.abs(d.gradient(u)).max()))             # 1.2e-13   (exact: 25)
print(float(jnp.abs(d.inverse_laplacian(u)).max()))    # 0.00160   (exact)
print(float(jnp.abs(d.laplacian(d.inverse_laplacian(u))).max()))   # 9.6e-15  (should be 1)
```

## Root cause

- `spectraldiffx/_src/fourier/operators.py:75` (1D uses `grid.k_dealias`), and the `* dealias_filter` factors in the 2D/3D linear operators.
- `operators.py:150-164, 287-301, 497-511` (`inverse_laplacian`) and `:342-378, 555-579` (`project_vector`) use the unmasked wavenumbers.

## Proposed fix

Preferred: linear operators never mask. `apply_dealias` stays available, `jacobian`/`advection_scalar` keep truncating products (see the `advection_scalar` issue in this epic). If some users rely on the mask-on-derivative behaviour, add an explicit `dealias: bool = False` keyword to the linear operators rather than tying it to the grid field.

Whichever is chosen, make it uniform: every operator on the class must treat the mask the same way, and `inverse_laplacian` must be the exact inverse of `laplacian` on the modes the class keeps.

Then update `navier_stokes_2d.py` and `qg_model.py` to use `deriv.laplacian` / `deriv.hyperviscosity` instead of the hand-rolled `K2` terms, and remove the "would cause NaN" comments.

## Acceptance criteria

- [ ] Test: `laplacian(inverse_laplacian(u)) == u` to 1e-10 for `u = cos(25x)` on the default N = 64 grid (1D, 2D, 3D).
- [ ] Test: `gradient(cos(kx))` is exact for every `k < N/2` on the default grid, not just `k ≤ N/3`.
- [ ] `project_vector` and the linear operators agree on which modes they keep.
- [ ] The three notebooks use the same idiom for the diffusion term.
- [ ] Docstrings state explicitly which methods truncate.
